src/entities/player.lua

```diff
@@ -120,4 +155,48 @@ function rotateVector(x, y, angle)
   return x*c - y*s, x*s + y*c
 end
 
+function player:beginAccelerate()
+  self.isaccelerating = true
+  self.isreversing = false
+end
+
+function player:beginBrake()
+  self.isbraking = true
+end
+
+function player:beginReversing()
+  self.isaccelerating = false
+  self.isreversing = true
+end
+
+function player:beginTurningLeft()
+  self.isturningleft = true
+  self.isturningright = false
+end
+
+function player:beginTurningRight()
+  self.isturningleft = false
+  self.isturningright = true
```

Comment:
If these handlers limit themselves to updating state about the control they represent, we could do things in the update loop such as going straight when both left and right are pressed, and thereafter if only left is released, we would know we could turn right.

Reply:
Good call, I'll adjust that.